### PR TITLE
Auto-detect graphics backend and add default to Cargo.toml

### PR DESCRIPTION
**Pull request by AidaPaul**
This is a small addition that should help improve new user experience. I explained the issue in the main project, hope that's okay! If not, let me know, I will recreate it here :)

https://github.com/amethyst/amethyst/issues/2092#issuecomment-573169126

### CODE REVIEW

**Review by azriel91 — APPROVED** [review]
Thank you :v:

**Comment by 0x6273** [comment]
I think this is a bad approach. Projects that are created on a mac should be able to compile without hassle on a pc, and vice-versa, and setting a default backend kinda breaks that. The default can't be overridden by just specifying `--features other-backend`, since features are additive, and having multiple backend features enabled doesn't compile.

Maybe having things just work by default for most users is worth the cost of requiring some users to either edit the Cargo.toml or specify `--no-default-features` and then re-enabling all the default features. But I think it would be much better to determine the desired backend at compile time instead, either by providing `amethyst build/run` commands that work as a wrapper to the cargo commands, or with a change to cargo itself.

**Comment by azriel91** [comment]
> Maybe having things just work by default for most users is worth the cost of requiring some users to either edit the Cargo.toml

That was my train of thought for people trying out amethyst (let it "just work" on first try). The ideal case is the binary can decide at runtime which backend to use. While we aren't there, perhaps a good alternative is, choosing the dev's OS is the default, with an option to not have a default. (am also happy with empty default, and in the output after generating the project, indicate that the feature needs to be specified in `cargo run --features <feature>`.)

**Comment by AidaPaul** [comment]
My reasoning on this is akin to make it as humanely as easy as possible for new users. Once they have their feet wet, did some graphics, got hello world running, etc, and want to share the code then they can always go to `Cargo.toml` and delete the defaults. By then they are already hooked, and will probably understand why it is like it is. 
And I see people from the other point on discord too where they want a release that will work on both Mac and Windows, but the few that came through also were not too thrilled by having to explain to "the others" to pass that specific command line argument whenever they try to run it. I think most of them settled as "default" and just instructions on how to disable that on mac/windows/Linux etc.

Maybe a sensible middle ground will be to:
1. Provide a parameter for `new` that will disable the auto-selection (aka for the not-so-newbies, or wanting less handholding, probably worth a paragraph in the beginners part of the book)
2. When generated with the selected engine, also put a comment on the line explaining why it's there, how to disable it via command line, or by removing it from `Cargo.toml`. In essence, giving a clear guidance to any person who will look at the `Cargo.toml` on this peculiarity, so they will know if they should handle it, and if they do, how.

I think this way would cover all the bases? Something like `amethyst new --no-defaults` for the more adventurous/experienced, and a well-documented default for the rest.

**Comment by AidaPaul** [comment]
I've added the suggested changes, example cargo.toml from `amethyst new testt`

```
[package]
name = "testt"
version = "0.1.0"
authors = []
edition = "2018"

[dependencies]
amethyst = "0.12.0"

[features]
# This sets default graphics backend for your project, it was automatically detected based
# on your operating system. It's a great starting point into your journey with amethyst,
# but if you wish to share this with other people, you may wan to consider removing it
# and appending `--feature vulkan` to your commands, letting people on
# different platforms to chose different ones.
# Alternatively you can leave the default on, and other users can still disable it by
# adding `--no-default-features` to their commands, followed by `--feature` of their choice.
#
# To read more about it, check out documentation available at:
# https://book.amethyst.rs/stable/appendices/c_feature_gates.html?highlight=vulkan#graphics-features
default = ["vulkan"]
empty = ["amethyst/empty"]
metal = ["amethyst/metal"]
vulkan = ["amethyst/vulkan"]
```

File: src/main.rs [hunk 1]
```diff
@@ -54,10 +60,15 @@ fn exec_new(args: &ArgMatches) {
         .expect("Bug: project_name is required");
     let project_name = project_name.to_owned();
     let version = args.value_of("amethyst_version").map(ToOwned::to_owned);
+    let no_defaults = match args.occurrences_of("no_defaults") {
```
Comment (0x6273) [inline src/main.rs]: This should be `args.is_present("no_defaults")`

Comment (AidaPaul) [reply src/main.rs]: Oh, you are right, this is much better! Fixing :)

**Review by 0x6273 — APPROVED** [review]

**Review by azriel91 — APPROVED** [review]
:v:

**Review by topokel — APPROVED** [review]

**Comment by jaynus** [comment]
r+

**Comment by jaynus** [comment]
Damn you keyboard.

bors r+

nm i guess i dont have permission here either.

**Pull request merged**